data_preprocessing/preprocess_code2seq_data.py

```diff
@@ -137,8 +146,11 @@ def convert_holdout(holdout_name: str, vocab: Vocabulary, config: PreprocessingC
         _ = [_ for _ in tqdm(results, total=n_buffers)]
 
 
-def preprocess(config: PreprocessingConfig, is_vocab_collected: bool, n_jobs: int):
+def preprocess(problem: str, data: str, is_vocab_collected: bool, n_jobs: int):
     # Collect vocabulary from train holdout if needed
+    config_function = _config_switcher[problem]
```

Comment:
You should check the dict for the key and raise an error if the user passed an unknown problem.
```
if problem not in _config_switcher:
    raise
...
```

Reply:
But as far as I understood it is already checked by `argparse` https://docs.python.org/3/library/argparse.html#choices